Modified SalesforceClient, HttpClient and API default version.

lventre opened the pull request:
**SalesforceClient**: used SecurityProtocolType.Tls12 as TLS 1.0 is not accepted anymore by Salesforce.
**HttpClient**: removed several using blocks that swallow exception traces + implemented IDisposable pattern.
Soap and Bulk request managers: **API default version** set to 38.0.

lventre committed:
SalesforceClient: used SecurityProtocolType.Tls12 as TLS 1.0 is not accepted anymore by Salesforce.
HttpClient: removed 'using' blocks that swallow exception traces + implemented IDisposable pattern.
Soap and Bulk request managers: API default version set to 38.0

lventre commented:
Michael,

This is my first pull request, so I hope I'm doing it well.

My first update is because TLS 1.0 not allowed anymore by Salesforce, so I replaced it by TLS 1.2.
I also updated the HttpClient classe because the PerformRequest method threw a NullReferenceException.  I had to remove using blocks  to retrieve the exception trace. I also took the opportunity to implement the IDisposable pattern.

Thank you for your great library!
Loïc

michael-shattuck commented on SalesforceMagic/Http/HttpClient.cs:
```diff
@@ -128,11 +148,26 @@ private void AddRequestBody(ref HttpWebRequest webRequest, HttpRequest request)
+            }
+
+            // Free any unmanaged objects here.
+            //
```
Let's remove these comments and the extra if statement for now. If we need it in the future we can add it back.

michael-shattuck reviewed (changes requested):
Thank you for the pull request. I added a comment to the implementation of IDisposable. If you could remove the items mentioned I'll merge it in. Awesome work!

lventre committed:
HttpClient: removed unnecessary comments.
SalesforceClient: removed duplicate code and added the possibility to change the security protocol.
The security protocol is a new constructor's argument with a default value. It is also added as the last parameter of the 'ChangeEnvironment' method to have the least impact on existing client code.

lventre committed:
Forgot to pass the security protocol argument to the first constructor.

lventre commented:
Hello,

I removed to unnecessary comments in HttpClient as requested.
I also found valuable to define one's own security protocol when using the library. Hence I added a new argument to SalesforceClient with a default value to have no impact on existing code (adding this argument to the Login method would have impacts on GetSession and GetSessionId methods too).

Looking forward to hearing from you,
Loïc

michael-shattuck commented:
This looks great thank you
